### ISSUE TITLE
Bug in Gibbs Sampling

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#660]
``` python
In [14]: >>> from pgmpy.factors import TabularCPD
    ...: >>> from pgmpy.models import BayesianModel
    ...: >>> intel_cpd = TabularCPD('intel', 2, [[0.7], [0.3]])
    ...: >>> sat_cpd = TabularCPD('sat', 2, [[0.95, 0.2], [0.05, 0.8]], evidence=['intel'], evidence_card=[2])
    ...: >>> student = BayesianModel()
    ...: >>> student.add_nodes_from(['intel', 'sat'])
    ...: >>> student.add_edge('intel', 'sat')
    ...: >>> student.add_cpds(intel_cpd, sat_cpd)
    ...: 
    ...: >>> from pgmpy.inference import GibbsSampling
    ...: >>> gibbs_chain = GibbsSampling(student)

In [15]: gibbs_chain.sample(size=1000)
Out[15]: 
     intel  sat
0        0    1
1        1    1
2        1    1
3        1    1
4        1    1
5        1    1
6        0    0
7        0    0
8        0    0
9        0    0
10       0    0
11       0    0
12       0    0
13       0    0
14       0    0
15       1    1
16       1    1
17       1    1
18       1    1
19       1    1
20       1    1
21       1    1
22       1    0
23       0    0
24       0    1
25       1    0
26       0    0
27       1    1
28       1    1
29       1    1
..     ...  ...
970      0    0
971      0    0
972      0    0
973      0    0
974      0    0
975      0    0
976      0    0
977      1    1
978      1    1
979      1    1
980      1    1
981      1    1
982      1    1
983      1    1
984      1    1
985      1    1
986      1    1
987      1    1
988      1    1
989      1    1
990      1    1
991      1    1
992      1    1
993      0    0
994      0    0
995      0    0
996      0    0
997      0    0
998      0    0
999      1    1

[1000 rows x 2 columns]

In [16]: gibbs_chain.sample(size=1000)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-16-e2a824d3f117> in <module>()
----> 1 gibbs_chain.sample(size=1000)

/home/ankur/pgmpy/pgmpy/inference/Sampling.py in sample(self, start_state, size)
    354 
    355         sampled = DataFrame(index=range(size), columns=self.variables)
--> 356         sampled.loc[0] = [st for var, st in self.state]
    357         for i in range(size - 1):
    358             for j, (var, st) in enumerate(self.state):

TypeError: 'NoneType' object is not iterable
```

**Comment by khalibartan** [comment]
Sending in a PR.

**Comment by saivig** [comment]
@khalibartan change similarly in self.generate_sample method also at [L391](https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/inference/Sampling.py#L391) even though logically it's not necessary as we are generating sample only for once but will be better if we follow as in self.sample method

**Comment by khalibartan** [comment]
Fixed in #662 and #719
